pkg/nfd-master/nfd-master.go

```diff
@@ -605,20 +604,81 @@ func (m *nfdMaster) nfdAPIUpdateOneNode(nodeName string) error {
 	return nil
 }
 
-func (m *nfdMaster) refreshNodeFeatures(cli *kubernetes.Clientset, nodeName string, annotations, labels map[string]string, features *nfdv1alpha1.Features) error {
+// validateExtendedResources validates extended resources and returns a map
+// of valid extended resources.
+func (m *nfdMaster) validateExtendedResources(features *nfdv1alpha1.Features, extendedResources ExtendedResources) ExtendedResources {
+	outExtendedResources := ExtendedResources{}
+	deniedNs := map[string]struct{}{"kubernetes.io": {}}
+	deniedWildCarNs := map[string]struct{}{".kubernetes.io": {}}
+	allowedNS := map[string]struct{}{nfdv1alpha1.ExtendedResourceNs: {},
+		nfdv1alpha1.FeatureLabelNs: {}}
+	for extendedResource, capacity := range extendedResources {
+		// Add possibly missing default ns
+		extendedResource = addNs(extendedResource, nfdv1alpha1.ExtendedResourceNs)
+		// Check if given NS is allowed
+		ns, _ := splitNs(extendedResource)
+		if isNamespaceDenied(ns, deniedWildCarNs, deniedNs) {
+			if _, ok := allowedNS[ns]; !ok {
```

Comment:
I think we should also allow sub-ns of nfdv1alpha1.ExtendedResourceNs, i.e.
`HasPrefix(ns, nfdv1alpha1.FeatureLabelSubNsSuffix`)